include/mbedtls/config.h

```diff
@@ -145,6 +146,7 @@
  *
  * Comment if your system does not have a correct clock.
  *
+ * \warning You should comment this out if the system's clock could be controlled by an attacker.
```

Comment:
Ok, as Gilles points out, things are a bit more complex, and if we put this warning we should probably also put a warning that this option also prevents time-based expiration of certificates. Perhaps improving this documentation would be best done in another PR. (Or if done in this PR, it should definitely be a separate commit.)